tests/cases/py_client/cli_test.py

```diff
@@ -267,6 +268,23 @@ def testUploadDownload(self):
                          downloadDir), username='mylogin', password='password')
         self.assertEqual(ret['exitVal'], 0)
 
+        # Test uploading with reference
+        queryList = []
+
+        @httmock.urlmatch(netloc='localhost', path='/api/v1/file$', method='POST')
+        def checkParams(url, request):
+            # Add query for every file upload request
+            queryList.append(six.moves.urllib.parse.parse_qs(url[3]))
+
+        with httmock.HTTMock(checkParams):
+            ret = invokeCli(
+                args + ['--reference', 'reference_string'], username='mylogin', password='password')
+
+        # Test if reference is sent with each file upload
+        for query in queryList:
```

Comment:
This loop body won't be entered if queryList remains empty, so we should also add an assertion of the expected length of queryList.